### ISSUE TITLE
Node teaser heading size is set incorrectly on first cached render

### ISSUE THREAD

**Issue by richardbporter** [uiowa/uiowa#3558]
# Problem
The node teaser heading size is set incorrectly on first cached render for bundles that implement the headline field.

# Steps to reproduce
Add the Articles block to a page and enter a title and set the heading size to h3 for example. Inspect the LB edit page markup and see that child elements are set to h4 which is correct. Save the page and inspect the same markup and see that the child elements are set to h2 which is incorrect.

If you edit the block and change the heading size from h3 to h4 for example, then child elements will be set to h5.

I think this problem is due to the fact that the block render cache tags do not yet exist the first time it is added to a page. That means we aren't really adding anything to the entity cache tags the first time:

https://github.com/uiowa/uiowa/blob/fa510cd3d76680f68bf51a09d04ec60b4c7d9702/docroot/modules/custom/sitenow_articles/sitenow_articles.module#L507-L512

We use this code in multiple places - person, articles, featured content, etc. After saving the page and editing in LB again, the cache tags are populated with `block_content:244` for example. That cache tag does not exist the first time.

# Proposed solution
TBD.

**Comment by cory-skeers** [comment]
It's been discussed, but confirming and noting: views-based list blocks (people currently, more to follow) do not have this issue.

**Comment by cory-skeers** [comment]
Note for when investigating/experimenting: The child elements on a new articles block may render incorrectly as other heading levels, possibly the "correct" one (say, correctly set to `h3` as they should, but through the render cache and not the settings from the block headline field).

**Comment by cory-skeers** [comment]
There's an additional issue if a node appears in multiple places.

Add an articles block, featured content block, etc. on the same or different pages, and set the headlines to different levels. The heading levels will potentially all update to the last block saved. Once any block is updated, it should invalidate the render caches for its results, but the newly cached version may then be used in all block instances.

## Steps to reproduce
- Create two articles blocks with different child heading sizes, save
- Inspect heading sizes for both blocks on regular page view--they likely match, meaning one is correct and one is incorrect
- Update one of the blocks
- Inspect heading sizes for both blocks on regular page view--they likely match, and both set to the heading level of the updated block

**Comment by richardbporter** [comment]
I'm having trouble reproducing the bug as initially described. I wonder if what I was experiencing was actually what you described in https://github.com/uiowa/uiowa/issues/3558#issuecomment-868700956

**Comment by joewhitsitt** [comment]
Got sidetracked at the moment but wanted to mention this. If:

>views-based list blocks (people currently, more to follow) do not have this issue.

then I started with the featured content block. ~~What I saw was that a new featured content block on a new page with an article and page. The page had the correct heading and the article didn't.~~ _- This turned out not to be the case and matches https://github.com/uiowa/uiowa/issues/3558#issuecomment-868700956_ I was going to to start digging to see the differences between the two renders and see if there was anything that could just be scoped differently.
